source/RedisInside/Redis.cs

```diff
@@ -93,13 +94,19 @@ public void Dispose()
 
         private class Config : IConfig
         {
+            private static readonly ConcurrentDictionary<int, byte> usedPorts = new ConcurrentDictionary<int, byte>();
             private static readonly Random random = new Random();
             internal Action<string> logger;
             internal int port;
 
             public Config()
             {
-                port = random.Next(49152, 65535 + 1);
+                do
+                {
+                    port = random.Next(49152, 65535 + 1);
+                } while (usedPorts.ContainsKey(port));

```

Comment:
Won't it make more sense to just keep hammering `while(!usedPorts.TryAdd(port, i => byte.MinValue, (i, b) => byte.MinValue);) { // etc } `. Cleaner and a tiny bit more thread-safe.